akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -929,7 +929,22 @@ final class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends
    * '''Cancels when''' downstream cancels
    */
   def merge[T >: Out](that: Graph[SourceShape[T], _]): javadsl.Flow[In, T, Mat] =
```

Comment:
well, it does: the «completes when» clause clearly describes the behavior, that’s good enough for me